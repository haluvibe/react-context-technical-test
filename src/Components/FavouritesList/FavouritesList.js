import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';
import './FavouritesList.css';

const FavouritesList = ({ favourites, removeFromFavourites }) => (
  <ul>
    {favourites.map(({ name, id }) => {
      return (
        <li key={uuid.v4()}>
          {name}
          <button className="remove-button" onClick={() => removeFromFavourites(id)}>
            {' '}
            Remove from Favourites{' '}
          </button>
        </li>
      );
    })}
  </ul>
);

FavouritesList.propTypes = {
  favourites: PropTypes.arrayOf(PropTypes.object).isRequired,
  removeFromFavourites: PropTypes.func.isRequired,
};

export default FavouritesList;
