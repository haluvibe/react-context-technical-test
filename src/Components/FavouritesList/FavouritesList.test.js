import React from 'react';
import { shallow } from 'enzyme';
import FavouritesList from './FavouritesList';

const favourites = [
  {
    id: 81,
    name: 'India Session Lager - Prototype Challenge',
  },
];

const removeFromFavourites = jest.fn();

const mockProps = {
  favourites,
  removeFromFavourites,
};

const mockUuidKey = '00000000-0000-0000-0000-000000000000';

jest.mock('uuid/v4', () => () => mockUuidKey);

const setUp = (props = mockProps) => {
  const wrapper = shallow(<FavouritesList {...props} />);
  return wrapper;
};

describe('FavouritesList', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setUp();
  });

  it('should render correctly with mock props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders list items mapped from props correctly', () => {
    const texts = wrapper.find('ul').map(node => node.text());
    expect(texts).toEqual(['India Session Lager - Prototype Challenge - ']);
    expect(wrapper.find('ul').exists()).toEqual(true);
    expect(wrapper.find('li').length).toBe(1);
  });
});
