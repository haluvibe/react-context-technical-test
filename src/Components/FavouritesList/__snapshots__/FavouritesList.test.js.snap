// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesList should render correctly with mock props 1`] = `
<ul>
  <li
    key="00000000-0000-0000-0000-000000000000"
  >
    India Session Lager - Prototype Challenge
    <button
      onClick={[Function]}
    >
       - 
    </button>
  </li>
</ul>
`;
